src/Storages/ObjectStorage/StorageObjectStorage.cpp

```diff
@@ -101,28 +102,41 @@ StorageObjectStorage::StorageObjectStorage(
     , distributed_processing(distributed_processing_)
     , log(getLogger(fmt::format("Storage{}({})", configuration->getEngineName(), table_id_.getFullTableName())))
 {
-    try
+    if (configuration->partitioning_style != "auto" && configuration->withPartitionWildcard())
     {
-        if (!lazy_init)
+        throw Exception(ErrorCodes::BAD_ARGUMENTS, "The {} macro can't be used with {} partitioning style",
+                        PartitionedSink::PARTITION_ID_WILDCARD, configuration->partitioning_style);
+    }
+
+    bool do_lazy_init = lazy_init && !columns_.empty() && !configuration->format.empty();
```

Comment:
This is awkward, probably something related to backporting thispr from upstream unmerged pr